lib/rules/prefer-arrow-callback.js

```diff
@@ -246,7 +249,37 @@ module.exports = {
                     !scopeInfo.super &&
                     !scopeInfo.meta
                 ) {
-                    context.report(node, "Unexpected function expression.");
+                    context.report({
+                        node,
+                        message: "Unexpected function expression.",
+                        fix(fixer) {
+
+                            // FIXME: Is there a better way to detect parser support for arrow functions?
+                            if (!context.parserOptions || !context.parserOptions.ecmaVersion || context.parserOptions.ecmaVersion < 6) {
```

Comment:
It's unnecessary, I think.
Developers always can disable this rule.

Other ES2015 rules (e.g. `no-var`, `prefer-const`) also don't check the language version.